### PR TITLE
Answer questions related to Mongoose

### PR DESCRIPTION
**Pull request by wilsonkaya**

### CODE REVIEW

**Comment by wilsonkaya** [comment]
After I made the pull request I have realized that the the test is not given the right answers and then I have rechecked my code. Non of my solutions are matching the necessities of diagnostic. I didn't pay enough attention what is being asked and they are not consoling "not found". Further more in my index I have tried implementing the code that we have used in previous training which is not necessary.

File: diagnostic.js [hunk 1]
```diff
@@ -32,25 +32,73 @@ const Movement = require('./models/movement.js');
...
+    console.error(error);
+  }).then(done);
+};
 // Success -> console.log new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  let search = {};
+  if (arguments[0] && arguments[1]) {
```
Comment (bernardlee) [inline diagnostic.js:52]: This entire conditional block is unnecessary.

File: diagnostic.js [hunk 2]
```diff
@@ -32,25 +32,73 @@ const Movement = require('./models/movement.js');
...
+    let criterion = arguments[1];
+    if (criterion[0] === '/') {
+      let regex = new RegExp(criterion.slice(1, criterion.length - 1));
+      search[field] = regex;
+    } else {
+      search[field] = criterion;
+    }
+  }
+
+  Movie.find(search).then(function(movie) {
```
Comment (bernardlee) [inline diagnostic.js:63]: The logic here is correct, but the model should be `Movement` not `Movie`.

File: diagnostic.js [hunk 3]
```diff
@@ -32,25 +32,73 @@ const Movement = require('./models/movement.js');
...
+    });
+  }).catch(console.error).then(done);
+};
 // Success -> console.log all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movie.findById(id).then(function(movie) {
+    console.log(movie.toObject());
```
Comment (bernardlee) [inline diagnostic.js:74]: This should invoke `toJSON()` not `toObject()`.  Again, the logic here is correct, but the model should be `Movement` not `Movie`.

File: diagnostic.js [hunk 4]
```diff
@@ -32,25 +32,73 @@ const Movement = require('./models/movement.js');
...
+    console.error(error);
+  }).then(done);
+};
 // Success -> If the specified Movement exists, console.log it as JSON;
 //              otherwise, console.log "Not Found" and exit.
 // Failure -> Console.error
 
-const update = (id, field, value) => {};
+const update = (id, field, value) => {
+  let modify = {};
```
Comment (bernardlee) [inline diagnostic.js:84]: This variable declaration is unnecessary since it never gets used in a meaningful way.

File: diagnostic.js [hunk 5]
```diff
@@ -32,25 +32,73 @@ const Movement = require('./models/movement.js');
...
+  modify[field] = value;
+  Movie.findById(id).then(function(movie) {
+    movie[field] = value;
```
Comment (bernardlee) [inline diagnostic.js:87]: The existence of the document should be checked before attempting to modify and save it.  This has the same issue as the two previous answers.

File: diagnostic.js [hunk 6]
```diff
@@ -32,25 +32,73 @@ const Movement = require('./models/movement.js');
...
+  }).catch(console.error).then(done);
+};
 // Success -> If the specified Movement exists, update it and console.log the
 //              updated Movement as JSON; otherwise, console.log "Not Found" and exit.
 // Failure -> Console.error
 
-const destroy = (id) => {};
+const destroy = (id) => {
+  Movie.findById(id).then(function(movie) {
+    return movie.remove();
```
Comment (bernardlee) [inline diagnostic.js:99]: This has the same issues as the previous answer.

**Comment by bernardlee** [comment]
1/5

**Pull request closed without merging**